File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/provider/StatusBuilder.java
package app.zxtune.fs.provider;

import android.database.Cursor;
import android.database.MatrixCursor;

class StatusBuilder {
  static Cursor makeError(Exception e) {
    final MatrixCursor result = new MatrixCursor(Schema.Status.COLUMNS, 1);
    result.addRow(Schema.Status.makeError(e));
    return result;
  }

  static Cursor makeProgress(int done) {
    final MatrixCursor result = new MatrixCursor(Schema.Status.COLUMNS, 1);
    result.addRow(Schema.Status.makeProgress(done));
    return result;
  }

  static Cursor makeProgress(int done, int total) {
    final MatrixCursor result = new MatrixCursor(Schema.Status.COLUMNS, 1);
    result.addRow(Schema.Status.makeProgress(done, total));
    return result;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/provider/StatusBuilder.kt
package app.zxtune.fs.provider

import android.database.Cursor
import android.database.MatrixCursor

internal object StatusBuilder {
    fun makeError(e: Throwable): Cursor = MatrixCursor(Schema.Status.COLUMNS, 1).apply {
        addRow(Schema.Status.Error(e).serialize())
    }

    fun makeProgress(done: Int, total: Int): Cursor = MatrixCursor(Schema.Status.COLUMNS, 1).apply {
        addRow(Schema.Status.Progress(done, total).serialize())
    }

    fun makeIntermediateProgress(): Cursor = MatrixCursor(Schema.Status.COLUMNS, 1).apply {
        addRow(Schema.Status.Progress.createIntermediate().serialize())
    }
}
